ocaml/database/db_cache_types.ml

```diff
@@ -26,14 +26,9 @@ end
 
 module StringPool = Weak.Make (HashedString)
 
-let pool = StringPool.create 2048
-
-let unique str =
-  match StringPool.find_opt pool str with
-  | None ->
-      StringPool.add pool str ; str
-  | Some s ->
-      s
+let share =
+  let pool = StringPool.create 2048 in
```

Comment:
Why has `pool` been moved from the module scope to function scope?

Reply:
I moved it because nobody else needs the pool. Is this wrong in the way that it would re-create the pool every time? My intention was to create a closure.

Reply:
Just tested and you're right, it creates the closure and pool gets reused correctly